zerver/tests/test_auth_backends.py

```diff
@@ -728,6 +738,33 @@ def test_google_oauth2_success(self):
         user_profile = get_user_profile_by_email('hamlet@zulip.com')
         self.assertEqual(get_session_dict_user(self.client.session), user_profile.id)
 
+    def test_google_oauth2_csrf_salt(self):
+        # type: () -> None
+        request = mock.MagicMock()
+        request.META = {}
+
+        # The csrf value should be deterministic.
+        result1 = google_oauth2_csrf(request, 'value')
+        result2 = google_oauth2_csrf(request, 'value')
+        self.assertEqual(result1, result2)
+
+        # If this is a version of Django that doesn't salt CSRF tokens (hence
+        # doesn't have the _unsalt_cipher_token function), the call to
+        # google_oauth2_csrf should handle the ImportError.
+        #
+        # Note: the try/except is necessary because __builtin__ is renamed to
+        # builtins in Python 3.
+        try:
+            with mock.patch(
+                    '__builtin__.__import__',
+                    side_effect=ImportError()):
+                google_oauth2_csrf(request, 'value') # nocoverage in Python 3
+        except ImportError:                          # nocoverage in Python 2
+            with mock.patch(                         # nocoverage in Python 2
+                    'builtins.__import__',           # nocoverage in Python 2
+                    side_effect=ImportError()):      # nocoverage in Python 2
+                google_oauth2_csrf(request, 'value') # nocoverage in Python 2
```

Comment:
We're using a fixed version of Django (1.10, used to be 1.8), I bet the code you're testing here was added as part of the transition to make things smooth.  So the right fix here is probably just to hardcode the Django 1.10 code path...